RELEASE.md

```diff
@@ -1,5 +1,32 @@
+# Release
+
 To release a new version of Spyder you need to follow these steps:
 
+## Localization updates
+
+* Install [gettext-helpers](https://github.com/spyder-ide/gettext-helpers) from source.
+
+* Update `*.pot` and `*.po` files to update translation strings by following these steps:
+  * Run `spyder-gettext scane spyder` to update localization files.
+  * Create and merge a new PR with these updated files
+  * Once merged, the new strings are now available on [Crowdin](https://crowdin.com/project/spyder).
+  * Request translators to update the translation strings on [Crowdin](https://crowdin.com/project/spyder). This process can take between a couple of days to a couple of weeks depending on the amount of strings to translate and we need to wait before proceeding to the next step.
+
+* Close the automatic translation bot PR `translate/<branch-name>`,
+* Delete the branch to recreate the branch automatically without conflicts (if any)
+  * Using this command: `git push origin --delete translate/<branch-name>`.
+  * Or, on the [github website](https://github.com/spyder-ide/spyder/branches)
+
+* Wait for the new `translate/<branch-name>` PR to be created by the `spyder-bot`. This branch should be created within 10 minutes.
+
+* Update `*.mo` files to update translation strings by following these steps:
+
+  * Run `spyder-gettext compile spyder` to update localization binary files.
+
+* Update the translation PR `translate/<branch-name>` to include these files and squash all commits into 1 single commit. These commits will include the `*.pot`, `*.po` and `*.mo` file changes.
```

Comment:
I assume `translate/<branch-name>` needs to be merged into `<branch-name>`. Perhaps should this be mentioned explicitly.

Reply:
Agreed!